checks/store/legacyrank/legacy_rank.go

```diff
@@ -0,0 +1,99 @@
+package legacyrank
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/csv"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+var ErrNotFound = errors.New("domain not found")
+
+type Getter interface {
+	GetLegacyRank(domain string) (int, error)
+}
+
+type GetterFunc func(domain string) (int, error)
+
+func (f GetterFunc) GetLegacyRank(domain string) (int, error) {
+	return f(domain)
+}
+
+type InMemoryStore struct{}
+
+var once sync.Once
+var data map[string]int //map of domain to rank
+
+func NewInMemoryStore() *InMemoryStore {
+	return &InMemoryStore{}
```

Comment:
Why would we return the actual InMemoryStore, instead of a pointer to it? Or would that just be unnecessary here?

Reply:
Pointer was as i intended to originally to have the data inside the struct. but then moved it to a package with a global for the one off load, either would work here, the difference is negligible although strictly speaking non pointer would be a tiny bit faster